action/governance/finalizeProposal.go

```diff
@@ -193,7 +193,12 @@ func runFinalizeProposal(ctx *action.Context, tx action.RawTx) (bool, action.Res
 func distributeFunds(ctx *action.Context, proposal *governance.Proposal, proposalDistribution *governance.ProposalFundDistribution) error {
 	// Required Perimeters for Fund Distribution
 	fundStore := ctx.ProposalMasterStore.ProposalFund
-	totalFunding := fundStore.GetCurrentFundsForProposal(proposal.ProposalID).Float()
+	totalFunds, err := fundStore.GetCurrentFundsForProposal(proposal.ProposalID)
+	if err != nil {
+		ctx.Logger.Error(err)
+		return err
+	}
+	totalFunding := totalFunds.Float()
```

Comment:
and we should not use floating number to do the calculation, instead we should use only BigInt, using MultiplyInt64 DivideInt64 in Coin object for example

Reply:
sure, will change that.

Reply:
The float function here returns big.float which is handled in the getPercentageCoin function ,although we are not dealing with accuracy while converting the float to float64 .

Reply:
since the Float() function here is defined as this, it's already big float.
```go
func (a Amount) Float() *big.Float {
	return new(big.Float).SetInt(a.BigInt())
}
```
and by mention coin and bigInt, do you want to me to change all parts in function distributeFunds to coin based calculation to preserve accuracy? thanks

Reply:
this function Float() is not the correct name, it should be BigFloat(). 
and we should not calculating using float64 at all. use the coin based calculation

Reply:
ok, then I'll change all parts in function distributeFunds to coin based calculation